avocado/core/test.py

```diff
@@ -630,6 +630,9 @@ def get_state(self):
         state['class_name'] = self.__class__.__name__
         state['job_logdir'] = self.job.logdir
         state['job_unique_id'] = self.job.unique_id
+        state['params'] = [(path, key, value)
+                           for path, key, value
+                           in self.params.iteritems()]
```

Comment:
@apahim @clebergnu hi, we got the following error while running vt tests, can you please help check it? thanks!
```
[stderr] Process Process-1:
[stderr] Traceback (most recent call last):
[stderr]   File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
[stderr]     self.run()
[stderr]   File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
[stderr]     self._target(*self._args, **self._kwargs)
[stderr]   File "/usr/lib/python2.7/site-packages/avocado_framework-59.0-py2.7.egg/avocado/core/runner.py", line 327, in _run_test
[stderr]     early_state = instance.get_state()
[stderr]   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-59.0-py2.7.egg/avocado_vt/test.py", line 183, in get_state
[stderr]     state = super(VirtTest, self).get_state()
[stderr]   File "/usr/lib/python2.7/site-packages/avocado_framework-59.0-py2.7.egg/avocado/core/test.py", line 647, in get_state
[stderr]     in self.params.iteritems()]
[stderr] ValueError: need more than 2 values to unpack
Avocado crashed: TestError: Process died before it pushed early test_status.
```

Reply:
@luckyh looks like this is related to Avocado-VT changing the meaning of `self.params` and calling Avocado code (`get_state()`).  This is a quick and dirty fix:

```diff
diff --git a/avocado_vt/test.py b/avocado_vt/test.py
index 639125143..170e67169 100644
--- a/avocado_vt/test.py
+++ b/avocado_vt/test.py
@@ -180,8 +180,10 @@ class VirtTest(test.Test):
         Avocado-vt replaces Test.params with avocado-vt params. This function
         reports the original params on `get_state` call.
         """
+        vt_params = self.params
+        self.params = self.avocado_params
         state = super(VirtTest, self).get_state()
-        state["params"] = self.avocado_params
+        self.params = vt_params
         return state
 
     def _start_logging(self):
```

Can you please check if this works for you? I'll think of a better fix in the morning.

Reply:
Yes, it works for me, thank you :)

Reply:
@clebergnu  this fix can fix the problem above, case can start running. but after finished, one meet another error:

 ```
avocado run --journal --vt-config /var/lib/avocado/data/avocado-vt/backends/qemu/cfg/tests-example.cfg
JOB ID     : 59f418d6ee05f339539b690240c2b4b6a084c35b
JOB LOG    : /root/avocado/job-results/job-2018-03-20T02.11-59f418d/job.log
 (1/1) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.boot: PASS (39.26 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 41.00 s
Error running method "render" of plugin "html": 'AvocadoParams' object is not iterable
```

could you help check if they are same issue ?